macros/src/derive_fn.rs

```diff
@@ -136,15 +145,50 @@ pub fn derive_fn(
         #link_section
         pub static #spec_ident: [u8; #spec_xdr_len] = *#spec_xdr_lit;
 
-        #export_name
-        pub fn #wrap_ident(env: stellar_contract_sdk::Env, #(#wrap_args),*) -> stellar_contract_sdk::RawVal {
-            <_ as stellar_contract_sdk::IntoVal<stellar_contract_sdk::Env, stellar_contract_sdk::RawVal>>::into_val(
-                #call(
-                    #env_call
-                    #(#wrap_calls),*
-                ),
-                &env
-            )
+        pub mod #mod_ident {
+            #export_name
+            pub fn call(env: stellar_contract_sdk::Env, #(#wrap_args),*) -> stellar_contract_sdk::RawVal {
```

Comment:
Could we name these functions `call_raw` and `call_raw_slice`? The term `call` on a function a developer has written may be confusing that it doesn't simply and only call it with the parameters I've defined. The `_raw` eludes a little to the parameters being raw vals, and makes it easier to add a `call` that simply calls through to their contract function, which I think we may want to add as well.

Reply:
I've made this change because I agree in principle that `call` not doing the obvious thing is surprising. But my general goal is that developers should never see these functions. This will already be true, for example, in rs-stellar-token-contract after this PR.